src/events/attributes.rs

```diff
@@ -791,6 +887,94 @@ mod xml {
     use super::*;
     use pretty_assertions::assert_eq;
 
+    #[cfg(any(doc, not(feature = "encoding")))]
+    mod attribute_value_normalization {
+
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        /// Empty values returned are unchanged
+        #[test]
+        fn test_empty() {
+            let raw_value = "".as_bytes();
+            let output = "".to_owned();
+            let attr = Attribute::from(("foo".as_bytes(), raw_value));
+            assert_eq!(attr.normalized_value(), Ok(Cow::Owned(output))); // TODO: this should not allocate
```

Comment:
A note: this does not check that value is owned, as was noted in https://github.com/tafia/quick-xml/pull/423#issuecomment-1407425485